tests/src/yaml-smoke.spec.ts

```diff
@@ -0,0 +1,77 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { Page } from '@playwright/test';
+import type { RunnerTestContext } from './testContext/runner-test-context';
+import { afterAll, beforeAll, test, describe, beforeEach, expect } from 'vitest';
+import { expect as playExpect } from '@playwright/test';
+import { PodmanDesktopRunner } from './runner/podman-desktop-runner';
+import { WelcomePage } from './model/pages/welcome-page';
+import { NavigationBar } from './model/workbench/navigation';
+import { deletePod } from './utility/operations';
+import path from 'path';
+
+let pdRunner: PodmanDesktopRunner;
+let page: Page;
+
+beforeAll(async () => {
+  pdRunner = new PodmanDesktopRunner();
+  page = await pdRunner.start();
+  pdRunner.setVideoName('play-yaml-e2e');
+
+  await new WelcomePage(page).handleWelcomePage(true);
+});
+
+beforeEach<RunnerTestContext>(async ctx => {
+  ctx.pdRunner = pdRunner;
+});
+
+afterAll(async () => {
+  await pdRunner.close();
+});
+
+describe.skipIf(process.env.GITHUB_ACTIONS && process.env.RUNNER_OS === 'Linux')(
+  'Play yaml file to pull images and create pods',
+  async () => {
+    test('Playing yaml', async () => {
+      const navigationBar = new NavigationBar(page);
+      let podsPage = await navigationBar.openPods();
+      await playExpect(podsPage.heading).toBeVisible();
+
+      const playYamlPage = await podsPage.openPlayKubeYaml();
+      await playExpect(playYamlPage.heading).toBeVisible();
+
+      const yamlFilePath = path.resolve(__dirname, '..', 'resources', 'primary-podify-demo.yaml');
+      podsPage = await playYamlPage.playYaml(yamlFilePath);
+      await playExpect(podsPage.heading).toBeVisible();
+
+      playExpect(await podsPage.podExists('podify-demo-pod')).toBeTruthy();
```

Comment:
here I would include test method:
so it would be like this;
```
test('Play yaml'){}
test('Primary-podify-pod is created') {}
test('Pod's images are present/pulled'){}
test('can delete pods and images'){}

Reply:
I'm not sure I understand what you mean, do you want to split the test in four parts?

Reply:
But if we do that then we're just testing the delete pods and images functionality and we're already doing that in another test. I'm not sure I see the point of such duplication. The way I understand it is that creating the pods and pulling the images are part of the "play yaml" functionality so a test should check that, but if you want to split it down further you are just retesting functionality that is already tested in another test. Deleting a pod from play yaml is the same as deleting a pod that was created in some other way.

Ultimately if we go with such atomization, we're going to end up with many very similar/identical tests in different test files.

Reply:
I agree with deleting part. It might be also part of clean up in after hook.
But as for checking images, pods and containers, I would stick to what was said. Since we are passing just a file with all inputs and it behaves as a blackbox with how is it processing the kube manifest (it just calls alternative of `podman play kube`). We need to make sure that correct images were pulled, containers in pod are named correctly (pod as well).

Reply:
I've split it into three tests:
1. play yaml
2. validate pod and delete pod
3. validate images and delete images